### PR TITLE
agentless: support updating health checks on consul clients during an upgrade to agentless

### PR DESCRIPTION
**Pull request by ishustava**
Changes proposed in this PR:
We will need to handle the case when a user has many applications on the service mesh and needs to take time to re-inject all those applications. If that is the case, there may be situations where the application changes its health status, but because it still hasn’t been re-started (re-injected), its health status needs to be synced into Consul clients rather than servers the way we used to do before. To do this, we will add a new annotation to all pods pointing to the current `consul-k8s version: consul.hashicorp.com/consul-k8s-version: 1.0.0`. If this annotation is not there or if the version is less than the version that supports “agentless”, we will continue to sync health checks to Consul clients for that particular pod

How I've tested this PR:
manual upgrade testing in multiple setups (not-secure, tls, tls +acls, tls+auto-encrypt+acls)

How I expect reviewers to test this PR:
👀 

Checklist:
- [x] Tests added
- [ ] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: charts/consul/templates/connect-inject-deployment.yaml [hunk 1]
```diff
@@ -236,24 +236,12 @@ spec:
                 {{- end }}
                 {{- end }}
 
-                {{- if .Values.global.consulSidecarContainer }}
```
Comment (ishustava) [inline charts/consul/templates/connect-inject-deployment.yaml:239]: These flags are no longer used

File: charts/consul/templates/connect-inject-deployment.yaml [hunk 2]
```diff
@@ -283,38 +271,38 @@ spec:
...
           {{- else }}
-          secretName: {{ template "consul.fullname" . }}-ca-cert
+            secretName: {{ template "consul.fullname" . }}-ca-cert
           {{- end }}
-          items:
-          - key: {{ default "tls.crt" .Values.global.tls.caCert.secretKey }}
-            path: tls.crt
+            items:
+              - key: {{ default "tls.crt" .Values.global.tls.caCert.secretKey }}
+                path: tls.crt
```
Comment (ishustava) [inline charts/consul/templates/connect-inject-deployment.yaml:305]: My IDE decided to format these

File: charts/consul/templates/connect-inject-deployment.yaml (hunk 1 above)
Comment (curtbushko) [reply charts/consul/templates/connect-inject-deployment.yaml:242]: The removal of these are actually in the PR I am working on right now as I missed them last week.

**Review by curtbushko — APPROVED** [review]
Nice job!

File: charts/consul/templates/connect-inject-deployment.yaml (hunk 1 above)
Comment (thisisnotashwin) [reply charts/consul/templates/connect-inject-deployment.yaml:242]: Hopefully the rebase of main onto your PR should automatically clear some of that from your commits.

**Review by thisisnotashwin — APPROVED** [review]
Excellent work, as usual 😄

**Pull request merged**